frontend/common/src/components/tabs/TabSet.tsx

```diff
@@ -0,0 +1,74 @@
+import React, { useState } from "react";
+import { TabProps } from "./Tab";
+
+export interface TabSetProps extends React.HTMLProps<HTMLElement> {
+  children: React.ReactElement<TabProps>[];
+}
+
+interface TabSetState {
+  selectedTab: number;
+  isOpen: boolean;
+}
+
+const firstSelectableTab = (tabs: React.ReactElement<TabProps>[]): number => {
+  const firstIndex = tabs.findIndex(
+    (tab: React.ReactElement<TabProps>) =>
+      tab.props.variant == null || tab.props.variant === "normal",
+  );
+  return firstIndex >= 0 ? firstIndex : 0;
+};
+
+export const TabSet: React.FC<TabSetProps> = ({
+  children,
+}): React.ReactElement => {
+  const [tabSetState, setTabSetState] = useState<TabSetState>({
+    selectedTab: firstSelectableTab(children),
+    isOpen: true,
+  });
+
+  const handleTabSelect = (index: number) => {
+    setTabSetState({
+      ...tabSetState,
+      isOpen: true,
+      selectedTab: index,
+    });
+  };
+  const handleToggleOpen = () => {
+    setTabSetState({
+      ...tabSetState,
+      isOpen: !tabSetState.isOpen,
+    });
+  };
+
+  // https://stackoverflow.com/a/32371612
+  const childrenWithProps = React.Children.map(children, (child, index) => {
+    // Checking isValidElement is the safe way and avoids a typescript
+    // error too.
+    if (React.isValidElement(child)) {
+      return React.cloneElement(child, {
+        isTabSetOpen: tabSetState.isOpen,
+        isTabSelected: tabSetState.selectedTab === index,
+        onSelect: () => handleTabSelect(index),
+        onToggleOpen: () => handleToggleOpen(),
+      });
+    }
+    return child;
+  });
+
+  return (
+    <>
+      <div
+        data-h2-display="b(flex)"
+        data-h2-align-items="b(center)"
+        data-h2-margin="b(bottom, s)"
+      >
+        {childrenWithProps}
+      </div>
+      <div style={{ display: tabSetState.isOpen ? "block" : "none" }}>
+        {children[tabSetState.selectedTab].props.children}
+      </div>
```

Comment:
With the Accordion, we decided to always render the children, even if they weren't displayed, so that they would retain any internal state. In this case, that means rendering the children of all tabs but keeping them hidden. Something like this:
```suggestion
      {children.map((child, index) => (
        <div style={{ display: tabSetState.isOpen && tabSetState.selectedTab === index ? "block" : "none" }}>
          {child.props.children}
        </div>
      ))}  
```

Reply:
This might break your jest tests?

Reply:
Ah, good thought.  I didn't consider the internal state of the tabs.